include/zenoh-pico/api/types.h

```diff
@@ -184,8 +184,8 @@ _Z_LOANED_TYPE(_z_queryable_t, queryable)
  * Represents a Zenoh Query entity, received by Zenoh Queryable entities.
  *
  */
-_Z_OWNED_TYPE_RC(_z_query_rc_t, query)
-_Z_LOANED_TYPE(_z_query_rc_t, query)
+_Z_OWNED_TYPE_PTR(_z_query_t, query)
```

Comment:
I just realized, that query likely should stay ref counted, because zenoh allows to make copies of queries and send multiple replies through them. The query is considered to be finalized when the last copy is dropped.